### PR TITLE
Javascript - Chapter

### PR DESCRIPTION
**Pull request by unrooted**
So, I created the part for JS, intro.md and README.md are created, check it out and merge if it's good enough.

### CODE REVIEW

File: book/languages/js/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Oh, hello there
+
+Welcome to the JavaScript chapter! I would like to let you know a bit more about JavaScript and programming in that language.
```
Comment (FilipeRamalho) [inline book/languages/js/README.md]: Readme.md is so fine.

File: book/credits.md [hunk 2]
```diff
@@ -34,4 +34,8 @@ We have to thank these great people on helping to write this book. Everyone here
 - gadilashashank
 
   - [Devrant](https://devrant.com/users/silverstr)
-  - [Github](https://github.com/gadilashashank)
\ No newline at end of file
+  - [Github](https://github.com/gadilashashank)
+
+- Konrad RootPixl Klawikowski
+  - [Devrant](https://devrant.com/users/RootPixl)
+  - [Github](https://github.com/RootPixl)
```
Comment (FilipeRamalho) [inline book/credits.md]: Changes to credits are so fine.

File: book/SUMMARY.md [hunk 3]
```diff
@@ -9,6 +9,9 @@
 * [C++](languages/cpp/README.md)
   * [Introduction](languages/cpp/intro.md)
 
+* [JS](languages/js/README.md)
+  * [Introduction](languages/js/intro.md)
+
 ## End
 
 * [Credits](credits.md)
```
Comment (FilipeRamalho) [inline book/SUMMARY.md]: Changes to credits are so fine.

File: book/languages/js/intro.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Introduction
+
+## Inception
+
+**JavaScript (JS)**, but the official name is **ECMAScript**. <br> This language is really fast scripting language using specially for web pages, for **frontend** (Vue.js, React.js etc.) and for **backend** (node.js etc.). It can be used also for creating desktop and mobile native apps (Proton, Electron).
+**Brednan Eich** created that language, and it firstly appeared in **December 4, 1995**. JS became an **ECMA standard in 1997**. <br> File extension is **.js**
+
```
Comment (FilipeRamalho) [inline book/languages/js/intro.md]: Inceptio is so fine.

File: book/languages/js/intro.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+## Trademark
+
+Actually Oracle have a trademark to JavaScript. Yes, these guys from Silicon Valley that are making virtual machines software.
+
```
Comment (FilipeRamalho) [inline book/languages/js/intro.md]: Not sure if the section Trademark is needed for just this Information.

File: book/languages/js/intro.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+| 1997          | ECMAScript 1  | First Edition |
+| 1998          | ECMAScript 2  | Editorial changes only |
+| 1999          | ECMAScript 3  | Added Regular Expressions, Added try/catch |
+|               | ECMAScript 4  | Was never released |
+| 2009          | ECMAScript 5  | Added "strict mode". Added JSON support. |
+| 2011          | ECMAScript 5.1| Editorial changes |
+| 2015          | ECMAScript 6  | Many new features (including let, const and default parameter values) |
+| 2016          | ECMAScript 7  | Added exponential operator (**), Added Array.prototype.includes  |
+| 2017          | ECMAScript 8  | Added string padding. Added new Object properies. Added Async functions. Added Shared Memory. |
+
```
Comment (FilipeRamalho) [inline book/languages/js/intro.md]: Not sure if there should be a Version table, maybe there should be instead a text About the changes and also it shouldn't get to technical

File: book/languages/js/intro.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+**You should use JavaScript** because:
+
+- Many of people actually use it, big community,
+- It is functionally in most of the browsers,
+- If you'll work as JS dev, you will be high paid one,
+- Many job offers in which JS is 'must have',
+- It is fast language,
+- It have many frameworks, I mean, **many frameworks**,
+- You could use it wherewhere you want.
+
```
Comment (FilipeRamalho) [inline book/languages/js/intro.md]: Add a text

File: book/languages/js/intro.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+## Why you shouldn't use JavaScript?
+
+**You shouldn't use JavaScript** because:
+
+- It is fast, but not as fast as C or Rust,
+- Many of people know it, it is good, but for searching one ideal job as a JS dev, it isn't,
+- It is nice to have a lot of frameworks, for backend and frontend, but the way, how much frameworks are for JS became a meme.
```
Comment (FilipeRamalho) [inline book/languages/js/intro.md]: Add a text

**Review by FilipeRamalho — CHANGES_REQUESTED** [review]
The Travis Build failed, so that has to be fixed.
There are mostly just technical details and basic facts. Build more on the story aspect. The files structure is like that ok. Also please less bias and extend the scope of the story to present day.

File: book/languages/js/intro.md (hunk 4 above)
Comment (gadilashashank) [reply book/languages/js/intro.md]: Removing inline html <br> for resolving travis error

File: book/languages/js/intro.md (hunk 8 above)
Comment (gadilashashank) [reply book/languages/js/intro.md]: Try to add the security aspect of JS and why it is a potential threat to website security

Comment (FilipeRamalho) [reply book/languages/js/intro.md]: That isn't true, you know ?

Comment (gadilashashank) [reply book/languages/js/intro.md]: @FilipeRamalho I beg to differ. Though JS has a strict mode it is not that widely implemented and JS can be used in attacking a client of a badly coded website. Though not the fault of JS it is related neverthless. 
https://stackoverflow.com/questions/3793246/javascript-security-risks

**Review by gadilashashank — COMMENTED** [review]
Some more info can be added some of which is mentioned in the comments both by me and Filipe. There are some grammatical errors which need to be corrected. Feel free to ask help regarding grammar.

File: book/languages/js/intro.md (hunk 8 above)
Comment (FilipeRamalho) [reply book/languages/js/intro.md]: Everything as risks, but that isn't a disadvantage. A disadvantage would be a security risk that is part of Javascript, but these are general web developement risks.

**Comment by FilipeRamalho** [comment]
You asked if something is missing, please read the review.

**Comment by FilipeRamalho** [comment]
There are mostly just technical details and basic facts. Build more on the story aspect. Also please less bias and extend the scope of the story to present day. 
Not sure if the section Trademark is needed for just only one fact. Not sure if there should be a Version table, maybe there should be instead a text About the changes and also it shouldn't get to technical. Add a text to advantages and disadavantages.


Review Grammar and focus on these aspects
- sentence structure
- spelling
- missing articles

**Comment by RootPixl** [comment]
Trademark section in this place is a good thing
Back to versions table, I think I could make versions.md and supported_browsers.md where I'll place that table
Grammar side should be fine for now, maybe one from Editors could change something
and I actually don't know more to add to story section except for ECMA Standard or mention some story from Mozilla, what do you think?

File: book/languages/js/intro.md (hunk 7 above)
Comment (Pavel-Durov) [reply book/languages/js/intro.md]: I would add:
+ Relatively easy language to learn
+ General purpose programming language - you can build Web/Mobile/Desktop/Server/VR/Game apps with it.

**Comment by Skayo** [comment]
Please split the intro.md into multiple files. 
The history part, for example, can definitely be put in a separate file.

**Comment by gadilashashank** [comment]
I'll take up your work and continue writing on top of it.

**Comment by gadilashashank** [comment]
I request you to check #33. If you wouldn't mind edits on JS can be continued there.

**Comment by gadilashashank** [comment]
I'll incorporate some sections of what you have written into #33 primarily because there are a tad too many grammatical errors and changes suggested till now. After I push them, we can possibly close this pull and continue working in #33.

**Comment by RootPixl** [comment]
@gadilashashank in my opinion we can continue work as well here and as well on [#33](https://github.com/devRant-Community/Community-Programming-Book/pull/33).
@FilipeRamalho in my opinion, that's kind of enough for JavaScript, check it out.

**Comment by gadilashashank** [comment]
Closing this. Further work will happen in #33. Though I have a feeling it is ready for merge.

**Pull request closed without merging**